wled00/wled.cpp

```diff
@@ -321,25 +320,33 @@ void WLED::setup()
 #endif
   DEBUG_PRINT(F("heap ")); DEBUG_PRINTLN(ESP.getFreeHeap());
 
-  #if defined(ARDUINO_ARCH_ESP32) && defined(WLED_USE_PSRAM)
-  if (psramFound()) {
-#if !defined(CONFIG_IDF_TARGET_ESP32C3) && !defined(CONFIG_IDF_TARGET_ESP32S2) && !defined(CONFIG_IDF_TARGET_ESP32S3)
-    // GPIO16/GPIO17 reserved for SPI RAM
-    managed_pin_type pins[2] = { {16, true}, {17, true} };
-    pinManager.allocateMultiplePins(pins, 2, PinOwner::SPI_RAM);
-#elif defined(CONFIG_IDF_TARGET_ESP32S3)
-    // S3: add GPIO 33-37 for "octal" PSRAM
-    managed_pin_type pins[5] = { {33, true}, {34, true}, {35, true}, {36, true}, {37, true} };
-    pinManager.allocateMultiplePins(pins, 5, PinOwner::SPI_RAM);
-#endif
-      DEBUG_PRINT(F("Total PSRAM: "));    DEBUG_PRINT(ESP.getPsramSize()/1024); DEBUG_PRINTLN("kB");
-      DEBUG_PRINT(F("Free PSRAM : "));    DEBUG_PRINT(ESP.getFreePsram()/1024); DEBUG_PRINTLN("kB");
-    } else
-      DEBUG_PRINTLN(F("No PSRAM found."));
+#if defined(ARDUINO_ARCH_ESP32) && defined(BOARD_HAS_PSRAM)
+  #if defined(CONFIG_IDF_TARGET_ESP32S3)
+  // S3: reserve GPIO 33-37 for "octal" PSRAM
+  managed_pin_type pins[] = { {33, true}, {34, true}, {35, true}, {36, true}, {37, true} };
+  pinManager.allocateMultiplePins(pins, sizeof(pins)/sizeof(managed_pin_type), PinOwner::SPI_RAM);
+  #elif defined(CONFIG_IDF_TARGET_ESP32S2)
+  // S2: reserve GPIO 26-32 for PSRAM
+  managed_pin_type pins[] = { {26, true}, {27, true}, {28, true}, {29, true}, {30, true}, {31, true}, {32, true} };
+  pinManager.allocateMultiplePins(pins, sizeof(pins)/sizeof(managed_pin_type), PinOwner::SPI_RAM);
+  #elif defined(CONFIG_IDF_TARGET_ESP32C3)
```

Comment:
Not sure if the pins for S2/C3 are compatible with pinMager.isPinOK rules.  Did you check that SPI_RAM pins actually get allocated?

allocateMultiplePins() does 
``` C++
if (!isPinOk(gpio, mptArray[i].isOutput)) 
   shouldFail = true;
[...]
if (shouldFail) return false;
```
isPinOk checks:
``` C++
 #if defined(CONFIG_IDF_TARGET_ESP32C3)
    // strapping pins: 2, 8, & 9
    if (gpio > 11 && gpio < 18) return false;     // 11-17 SPI FLASH
    if (gpio > 17 && gpio < 20) return false;     // 18-19 USB-JTAG
  [...]
  #elif defined(CONFIG_IDF_TARGET_ESP32S2)
    // strapping pins: 0, 45 & 46
    if (gpio > 21 && gpio < 33) return false;     // 22 to 32: not connected + SPI FLASH
    // JTAG: GPIO39-42 are usually used for inline debugging
    // GPIO46 is input only and pulled down
#endif
```

Reply:
Good catch!
Though the code may fail to allocate pins in current implementation, it may be future proof if we decide to allow `isPinOK()` to return `true` for SPI pins (as is the case for octal SPI on S3). There are no side effects when the allocation fails in case of `isPinOK()` returning `false`.
If allocation fails due to `isPinOK()` any and every subsequent attempt to allocate such pin will also fail. Regardless of the owner.